### DISCUSSION TITLE
TestNG and maybe use of Java's Virtual Threads

### DISCUSSION THREAD

**Discussion by dmirica-gpsw** [cbeust/testng#3142] [Ideas]
If I'm not mistaken TestNG doesn't use virtual threads when running tests in parallel. Using virtual threads would bring improvements? I'm thinking that at least some tests (eg selenium) are waiting on network which I think is one of the examples for virtual threads use.
Probably this will have implications regarding what java will be required and I assume that the implementation effort would be consistent.
I'm curios about other opinions and related info.

**Comment by krmahadevan** [comment]
@dmirica-gpsw - Yes you are right. TestNG doesn't use virtual threads yet, because its minimal JDK requirement is still `JDK11`. Maybe in the next major release of TestNG (say `v8.0.0`) we could move to doing something around that. But I don't have any concrete timelines as to when can that happen. 

The codebase may also require some refactoring around this to allow users to toggle between real and virtual threads preferably via a configuration. Please feel free to add more context, if you have done a bit more involved research around this topic.
